File: Chapter10/Python/Example2.py
# Chapter 10 - Δειγματοληπτικές Κατανομές και Κεντρικό Οριακό Θεώρημα
# Example 2: ΚΟΘ με Εκθετική Κατανομή (Exp λ=0.5)
#   Έντονα ασύμμετρη — n = 5, 10, 30, 100

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import norm, expon

np.random.seed(42)
n_simulations  = 10000
lambda_param   = 0.5
sample_sizes   = [5, 10, 30, 100]

# Population parameters
pop_mean = 1 / lambda_param   # = 2
pop_sd   = 1 / lambda_param   # = 2

fig = plt.figure(figsize=(15, 10))

# Plot original exponential distribution
ax1   = plt.subplot(2, 3, 1)
x_exp = np.linspace(0, 10, 200)
y_exp = expon.pdf(x_exp, scale=1 / lambda_param)
ax1.plot(x_exp, y_exp, 'b-', linewidth=2)
ax1.set_xlabel('x', fontsize=11)
ax1.set_ylabel('Density', fontsize=11)
ax1.set_title('Original Exponential Distribution', fontsize=12)
ax1.grid(alpha=0.3)

# Plot sampling distributions
for idx, n in enumerate(sample_sizes, start=2):
    sample_means = [np.mean(np.random.exponential(
                        scale=1 / lambda_param, size=n))
                    for _ in range(n_simulations)]

    se = pop_sd / np.sqrt(n)

    ax = plt.subplot(2, 3, idx)
    ax.hist(sample_means, bins=50, density=True,
            alpha=0.6, color='blue', edgecolor='black')

    x = np.linspace(min(sample_means), max(sample_means), 200)
    y = norm.pdf(x, loc=pop_mean, scale=se)
    ax.plot(x, y, 'r-', linewidth=2, label='Theoretical N')

    ax.axvline(pop_mean, color='green', linestyle='--',
               linewidth=2, label='Pop. Mean')

    ax.set_xlabel('Sample Mean', fontsize=11)
    ax.set_ylabel('Density', fontsize=11)
    ax.set_title(f'n = {n}', fontsize=12)
    ax.set_xlim(0, 5)
    ax.legend(fontsize=9)
    ax.grid(alpha=0.3)

plt.suptitle('CLT — Exponential Distribution (lambda=0.5)',
             fontsize=14, y=1.01)
plt.tight_layout()
plt.show()

File: Chapter10/R/Example2.R
# Chapter 10 - Δειγματοληπτικές Κατανομές και Κεντρικό Οριακό Θεώρημα
# Example 2: ΚΟΘ με Εκθετική Κατανομή (Exp λ=0.5)
#   Έντονα ασύμμετρη — n = 5, 10, 30, 100

set.seed(42)
n_simulations <- 10000
lambda        <- 0.5
sample_sizes  <- c(5, 10, 30, 100)

# Population parameters
pop_mean <- 1 / lambda  # = 2
pop_sd   <- 1 / lambda  # = 2

par(mfrow = c(2, 3))

# Plot original exponential distribution
x_exp <- seq(0, 10, length.out = 200)
y_exp <- dexp(x_exp, rate = lambda)
plot(x_exp, y_exp, type = "l", lwd = 2, col = "blue",
     main = "Original Exponential Distribution",
     xlab = "x", ylab = "Density")

# Plot sampling distributions for each sample size
for (n in sample_sizes) {
  sample_means <- replicate(n_simulations, {
    mean(rexp(n, rate = lambda))
  })

  se <- pop_sd / sqrt(n)

  hist(sample_means, breaks = 50, freq = FALSE,
       col    = "lightblue", border = "black",
       main   = paste("n =", n),
       xlab   = "Sample Mean",
       ylab   = "Density",
       xlim   = c(0, 5))

  x <- seq(min(sample_means), max(sample_means), length.out = 200)
  y <- dnorm(x, mean = pop_mean, sd = se)
  lines(x, y, col = "red", lwd = 2)

  abline(v = pop_mean, col = "darkgreen", lwd = 2, lty = 2)
}

par(mfrow = c(1, 1))
